### PR TITLE
Fix deprecated getEntity (fixes #67)

### LINKED ISSUE #67: WARNING: DraftEntity.get will be deprecated soon! Please use "contentState.getEntity" instead.
Related to https://github.com/sstur/draft-js-import-element/issues/31

This should be an easy fix.

### PR DESCRIPTION
**Pull request by mikaelwaltersson**
WARNING: DraftEntity.get will be deprecated soon! Please use "contentState.getEntity" instead.

### CODE REVIEW

**Review by Budaa — APPROVED** [review]

File: src/stateToHTML.js [hunk 1]
```diff
@@ -157,19 +156,19 @@ class MarkupGenerator {
...
+    }
+    this.contentState = contentState;
+    this.options = options;
+    let [inlineStyles, styleOrder] = combineOrderedStyles(
+      options.inlineStyles,
+      [DEFAULT_STYLE_MAP, DEFAULT_STYLE_ORDER],
+    );
+    this.inlineStyles = inlineStyles;
+    this.styleOrder = styleOrder;
+  }
```
Comment (sstur) [inline src/stateToHTML.js]: What's the difference between the code on the left and the code on the right? If there are no changes can you rebase this PR so they are the same?

File: src/stateToHTML.js [hunk 2]
```diff
@@ -418,6 +417,6 @@ function encodeAttr(text: string): string {
     .split('"').join('&quot;');
 }
 
-export default function stateToHTML(content: ContentState, options: ?Options): string {
-  return new MarkupGenerator(content, options).generate();
+export default function stateToHTML(content: ContentState, options: ?Options): string {
+  return new MarkupGenerator(content, options).generate();
```
Comment (sstur) [inline src/stateToHTML.js]: Also, what is the difference here?

File: src/stateToHTML.js [hunk 3]
```diff
@@ -341,7 +340,7 @@ class MarkupGenerator {
         }
         return content;
       }).join('');
-      let entity = entityKey ? Entity.get(entityKey) : null;
+      let entity = entityKey ? this.contentState.getEntity(entityKey) : null;
```
Comment (sstur) [inline src/stateToHTML.js]: This is the main change of this PR, right? This fixes the issue?

**Review by sstur — COMMENTED** [review]
Thanks for this contribution! I've got a few questions here..

File: src/stateToHTML.js (hunk 1 above)
Comment (mikaelwaltersson) [reply src/stateToHTML.js]: No change really, must be some spacing or something changed just be opening and saving the file in my editor. I couldn't manage to unstage the hunk either using SourceTree.

File: src/stateToHTML.js (hunk 2 above)
Comment (mikaelwaltersson) [reply src/stateToHTML.js]: No change really, must be some spacing or something changed just be opening and saving the file in my editor. I couldn't manage to unstage the hunk either using SourceTree.

File: src/stateToHTML.js (hunk 3 above)
Comment (mikaelwaltersson) [reply src/stateToHTML.js]: Correct, Entity.get is deprecated as of 0.10 and replaced by contentState.getEntity

Comment (mikaelwaltersson) [reply src/stateToHTML.js]: https://facebook.github.io/draft-js/docs/v0-10-api-migration.html#content

Comment (mikaelwaltersson) [reply src/stateToHTML.js]: I have removed the accidental trailing whitespace changes now.

**Comment by gbbr** [comment]
👍

**Comment by svnm** [comment]
Have tested and this works well for me in `Draft 0.10.0` 👍

**Comment by blade254353074** [comment]
How is it now?

**Comment by sstur** [comment]
Thanks for this contribution. My only concern is that this will NOT work with older versions draft-js which do not support `contentState.getEntity()`

We should create a helper function with feature detection:
```
function getEntity(contentState, entityKey) {
  return contentState.getEntity ? contentState.getEntity(entityKey) : Entity.get(entityKey);
}
```
Then you can change line 343 to:
```
let entity = entityKey ? getEntity(this.contentState, entityKey) : null;
```

**Review by sstur — DISMISSED** [review]
Thanks for this contribution. My only concern is that this will NOT work with older versions draft-js which do not support `contentState.getEntity()`

We should create a helper function with feature detection:
```
function getEntity(contentState, entityKey) {
  return contentState.getEntity ? contentState.getEntity(entityKey) : Entity.get(entityKey);
}
```
Then you can change line 343 to:
```
let entity = entityKey ? getEntity(this.contentState, entityKey) : null;
```

**Comment by mikaelwaltersson** [comment]
Added your improvements for backward compatibility

File: src/stateToHTML.js [hunk 4]
```diff
@@ -377,6 +376,10 @@ class MarkupGenerator {
 
 }
 
+function getEntity(contentState, entityKey) {
+  return contentState.getEntity ? contentState.getEntity(entityKey) : Entity.get(entityKey);
```
Comment (sstur) [inline src/stateToHTML.js:381]: `Entity` is not defined, is it?
I think because you removed line 7.

**Comment by sstur** [comment]
Please run `npm test` before submitting. This is partly my fault because I should setup TravisCI to automatically run test on all PRs. This is the error:

screenshot 2017-03-23 12 41 56

**Comment by mikaelwaltersson** [comment]
Sorry for that, weren't able to install all the tools necessary to run the tests and unfortunately I was to busy to look into it any further. 

screen shot 2017-03-23 at 4 59 46 pm

**Comment by sstur** [comment]
Thanks!

**Comment by Gm33** [comment]
Any news on this?

**Pull request merged**